lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -724,6 +725,103 @@ def cmd_services_help(default_columns)
+    else
+      ::ApplicationRecord.connection_pool.with_connection do |conn|
+        if conn.respond_to?(:current_database)
+          cdb = conn.current_database
+        end
+      end
+    end
+
+    if cdb.empty?
+      output = "Connected Database Name could not be extracted. DB Connection type: #{framework.db.driver}."
+    else
+      output = "Connected to #{cdb}. Connection type: #{framework.db.driver}."
+    end
+
+    output
+  end
+
+  def cmd_db_stats(*args)
+    return unless active?
+    print_line "Session Type: #{db_connection_info(framework)}"
+
+    current_workspace = framework.db.workspace
+    example_workspaces = ::Mdm::Workspace.order(id: :desc)
+    ordered_workspaces = ([current_workspace] + example_workspaces).uniq.sort_by(&:id)
+
+    tbl = Rex::Text::Table.new(
+    'Indent'  => 2,
+    'Header'  => "Database Stats",
+    'Columns' =>
+      [
+        "IsTarget",
+        "ID",
+        "Name",
+        "Hosts",
+        "Services",
+        "Services per Host",
+        "Vulnerabilities",
+        "Vulns per Host",
+        "Notes",
+        "Creds",
+        "Kerberos Cache"
+      ],
+    'SortIndex' => 1,
+    'ColProps' => {
+      'IsTarget' => {
+        'Stylers' => [Msf::Ui::Console::TablePrint::RowIndicatorStyler.new],
+        'ColumnStylers' => [Msf::Ui::Console::TablePrint::OmitColumnHeader.new],
+        'Width' => 2
+      }
+    }
+    )
+    total_creds = 0
+    total_tickets = 0
+    ordered_workspaces.map do |workspace|
+      total_creds += framework.db.creds(workspace: workspace.name).count
+      total_tickets += ticket_search([],nil, workspace).count
+      tbl << [
+        current_workspace.id == workspace.id,
+        workspace.id,
+        workspace.name,
+        workspace.hosts.count.to_fs(:delimited),
+        workspace.services.count.to_fs(:delimited),
+        workspace.hosts.count > 0 ? (workspace.services.count.to_f / workspace.hosts.count).truncate(2) : 0,
+        workspace.vulns.count.to_fs(:delimited),
+        workspace.hosts.count > 0 ? (workspace.vulns.count.to_f / workspace.hosts.count).truncate(2) : 0,
+        workspace.notes.count.to_fs(:delimited),
+        framework.db.creds(workspace: workspace.name).count.to_fs(:delimited), # workspace.creds.count.to_fs(:delimited) is always 0 for whatever reason
+        ticket_search([],nil, workspace).count.to_fs(:delimited)
+      ]
+    end
+
+    # total row
+    tbl << [
+      "",
+      "Total",
+      ::Mdm::Workspace.count.to_fs(:delimited),
+      ::Mdm::Host.count.to_fs(:delimited),
+      ::Mdm::Service.count.to_fs(:delimited),
+      ::Mdm::Host.count> 0 ? (::Mdm::Service.count.to_f / ::Mdm::Host.count).truncate(2) : 0,
+      ::Mdm::Vuln.count.to_fs(:delimited),
+      ::Mdm::Host.count > 0 ? (::Mdm::Vuln.count.to_f / ::Mdm::Host.count).truncate(2) : 0,
+      ::Mdm::Note.count.to_fs(:delimited),
+      total_creds.to_fs(:delimited),
+      total_tickets.to_fs(:delimited)
+    ]
+
+    puts tbl.to_s
```

Comment:
Using `puts` will print to `stdout`, but the console instance might be running under the context of RPC etc - which requires `print_line` or similar

```suggestion
print_line tbl.to_s
```